packages/code-studio/src/settings/FormattingSectionContent.tsx

```diff
@@ -40,44 +39,31 @@ import {
   isSameDecimalOptions,
   isSameIntegerOptions,
   isValidFormat,
-  removeFormatRuleExtraProps,
-  isFormatRuleValidForSave,
-  ValidFormatterItem,
 } from './SettingsUtils';
-import type { FormatterItem, FormatOption } from './SettingsUtils';
+import type { FormatOption } from './SettingsUtils';
 import DateTimeOptions from './DateTimeOptions';
 import TimeZoneOptions from './TimeZoneOptions';
 
 const log = Log.module('FormattingSectionContent');
 
 interface FormattingSectionContentProps {
   dh: DhType;
-  formatter: FormatterItem[];
   defaultDateTimeFormat: string;
   showTimeZone: boolean;
   showTSeparator: boolean;
   timeZone: string;
   truncateNumbersWithPound: boolean;
-  settings: WorkspaceSettings;
-  saveSettings: (settings: WorkspaceSettings) => void;
+  updateSettings: (settings: Partial<WorkspaceSettings>) => void;
   defaultDecimalFormatOptions: FormatOption;
   defaultIntegerFormatOptions: FormatOption;
-  defaults: {
-    defaultDateTimeFormat: string;
-    defaultDecimalFormatOptions: FormatOption;
-    defaultIntegerFormatOptions: FormatOption;
-    showTimeZone: boolean;
-    showTSeparator: boolean;
-    timeZone: string;
-  };
+  defaults: WorkspaceSettings;
 }
 
 interface FormattingSectionContentState {
-  formatSettings: FormatterItem[];
   showTimeZone: boolean;
   showTSeparator: boolean;
   timeZone: string;
-  defaultDateTimeFormat: string;
+  defaultDateTimeFormat?: string;
```

Comment:
Unsure why this is optional, it shouldn't be as we always have a default provided.
```suggestion
  defaultDateTimeFormat: string;
```